packages/flatbuffers/src/type-wrapper.ts

```diff
@@ -54,7 +56,43 @@ type UnwrappedType<T extends WrappedType> = T extends 'Bool'
   ? Buffer
   : never
 
-function unwrap<T extends WrappedType>(wrapper: TypeWrapper): UnwrappedType<T> {
+type RuleDouble =
+  | { type: 'number'; mode: 'absolute'; max?: number; min?: number }
+  | { type: 'number'; mode: 'relative'; max?: number; percent?: boolean }
+
+export type Rule = RuleDouble
+
+type UnwrappedRule<T extends WrappedType> = T extends 'Double'
+  ? RuleDouble
+  : never
+
+function makeRuleDouble(rule: ComparisonRuleDouble): RuleDouble {
+  const mode = rule.mode()
+  const max = rule.max() === null ? undefined : rule.max()!
+  const min = rule.min() === null ? undefined : rule.min()!
+  if (mode === ComparisonRuleMode.Absolute) {
+    return { type: 'number', mode: 'absolute', min, max }
+  }
+  const percent = rule.percent() === null ? undefined : rule.percent()!
+  return { type: 'number', mode: 'relative', max, percent }
+}
+
+export function unwrap_rule<T extends WrappedType>(
```

Comment:
I gave this a try but since we use `unwrapValue` recursively in Array and Objects, it felt a bit confusing. Would love to pair on this with you as we expand support for objects and arrays.